### ISSUE TITLE
Set up teach.mofodev.net

### ISSUE THREAD

**Issue by toolness** [mozilla/teach.webmaker.org#587]
I'm thinking we should probably set up a teach.mofodev.net that is separate from the existing teach.mofostaging.net.

In short, teach.mofostaging.net should really *become* teach.mofodev.net: it's deployed-to every time the `develop` branch changes, and its code is completely unminified and such.

There are then two options for teach.mofostaging.net, and they both rely on how we want to deploy to teach.mozilla.org (i.e. production):

* Deploy to mofostaging as soon as we tag a release and push it to `master` (which as I understand is part of this whole [git-flow](https://github.com/mozilla/teach.webmaker.org/issues/121) thing?). We can then use mofostaging to QA anything before *manually* pushing the release to teach.webmaker.org.
* Have tagging+pushing to `master` actually automatically push to teach.mozilla.org. teach.mofostaging.net will essentially just be the minified, production-ready version of teach.mofodev.net.

Any thoughts, @simonwex / @ScottDowne / et al? I am not sure what standard procedure for mofo is here.

**Comment by simonwex** [comment]
I'm on-board 100%. 

tl;dr for those dropping by this ticket: 
teach.mofodev.net: *The current state of work*
teach.mofostaging.net: *What's ready to go to production*

**Comment by jbuck** [comment]
Is there a meaningful difference between development and staging in this case? If the only practical difference is minification, then why not simplify our lives and always minify the code when building? If the source maps are external and not embedded then there's no penalty aside from the time needed to compress/minify js. Then we don't need two "pre-production" sites.

I'd also argue that develop vs master vs tag is an additional complexity that we can scrap, but what do you think about just having a "production" teach.mozilla.org and a "pre-production" teach.mofodev.net or teach.mofostaging.net ?

**Comment by toolness** [comment]
> Is there a meaningful difference between development and staging in this case

That's a good question--I *think* that there isn't much of one, aside from what git-flow prescribes, where "develop" is the wild west and "master" are the tagged releases that are ready to go live (but for which we might want a staging server to try them out on before pushing live). But that gets to your other point...

> I'd also argue that develop vs master vs tag is an additional complexity that we can scrap

I actually sort-of brought this up in https://github.com/MozillaFoundation/MoFo-Engineering-Handbook/issues/47#issuecomment-78375766, where I asked if we wanted to use Git**Hub** Flow rather than git-flow. My understanding is that the GitHub folks think git-flow is a bit overly complex in the same way that you describe, and that master is their live thing, and there is no develop.

**Comment by toolness** [comment]
Oh, also I think it would be neat to keep the dev ribbon around, since it provides a nice affordance for folks who are working on the site. I was thinking that could stay on dev but be kept off staging, but I dunno.

**Comment by toolness** [comment]
Ok, in 10e25f88d2f39a77dc3800bde59f12ec3cd0dfc9 I changed things so that the minified version of the site is being deployed to teach.mofostaging.net. It's not **identical** to production because it integrates w/ staging servers and it also has the dev ribbon, but in all other respects it is the same.

**Comment by toolness** [comment]
I think the current situation is probably good for now so I am closing this issue.
